### PR TITLE
feat(parish-mcp): player-triggered, MCP-readable screenshot capture

### PR DESCRIPTION
**Pull request by dmooney**
## Summary

Implements the player-triggered, MCP-readable screenshot scope from
`parish/crates/parish-mcp/README.md` (Future work → Screenshot capture).

Pressing **F2** in the live desktop window captures the Svelte UI via
`html-to-image`, the Tauri `save_screenshot` command decodes the PNG and
writes it to `<saves_dir>/screenshots/parish-<ISO-timestamp>.png`, and
the new `parish_latest_screenshot` MCP tool reads
`{path, taken_at, size_bytes}` back through the same in-process bridge
that already drives `parish_world_snapshot`, `parish_submit_input`, etc.

### What changed

- **Frontend** (`parish/apps/ui/`)
  - new `lib/screenshot.ts` with `captureScreen()` (`html-to-image`)
  - new `saveScreenshot` / `getLatestScreenshot` wrappers in `lib/ipc.ts`
  - F2 keybinding + transient bottom-centre toast in `routes/+page.svelte`
  - new dep: `html-to-image@^1.11.13`

- **Backend** (`parish-tauri`)
  - `save_screenshot(data_url) -> ScreenshotInfo` and
    `get_latest_screenshot() -> Option<ScreenshotInfo>` Tauri commands
    + `do_*` helpers shared with the bridge
  - new `AppState::latest_screenshot_path: Mutex<Option<PathBuf>>` field
  - `mcp_bridge.rs`: `GET /api/latest-screenshot`
  - registry + handlers updated; `base64` dep added

- **Server stubs** (`parish-server`)
  - 501 stubs for `POST /api/save-screenshot` and
    `GET /api/latest-screenshot` (Tauri-only feature, same pattern as
    the existing `demo_*` and `setup_*` stubs)
  - `route_registry.rs` updated → `wiring_parity` stays green

- **MCP** (`parish-mcp`)
  - `parish_latest_screenshot` tool (empty-input schema → routes to
    `get_latest_screenshot`)

- **Docs**
  - `parish-mcp/README.md`: tool table row added; "Future work" rewritten
    to reflect the implementation and the two design extensions still
    open (MCP-trigger; inline image-part response)
  - `AGENTS.md`: tool table row added

- **Proof bundle**: `docs/proofs/screenshot-mcp/{evidence.md,judge.md}`

### Tests

- 7 new Rust unit tests in `parish-tauri::commands::cmd_tests`
  cover base64 round-trip, filename formatting, AppState mutation,
  and the two missing-file edge cases for the reader.
- 2 new MCP tool tests pin translation + registry membership.
- The bridge route-table assertion now includes
  `/api/latest-screenshot` and `get_latest_screenshot`.
- 3 new frontend tests cover capture target selection and option
  forwarding via `vi.mock('html-to-image')`.
- `command_registry::command_count_matches_registry` bumped 32 → 34.

```
cargo test --workspace               all green (~1300 tests)
cargo clippy --all-targets -- -D warnings   clean
cargo fmt --all                      clean
npx vitest run                       399 passed
svelte-check                         0 errors
just agent-check                     proof + judge present
```

## Test plan

- [ ] Pull the branch and run `cargo test --workspace`.
- [ ] `cd parish/apps/ui && npm install && npx vitest run` → 399 passed.
- [ ] Launch the desktop with `cargo run -p parish-tauri -- --mcp-port 3030`,
      press **F2**, confirm the toast names a path under `saves/screenshots/`.
- [ ] From an MCP client: `parish_latest_screenshot` returns the same path
      and a non-zero `size_bytes`.
- [ ] In headless mode, `GET /api/latest-screenshot` returns 501 with the
      "Tauri-only" message — confirms mode parity is wired but inert.

https://claude.ai/code/session_017Pud6pVvhkGkHPNV1iSdeK

---
_Generated by [Claude Code](https://claude.ai/code/session_017Pud6pVvhkGkHPNV1iSdeK)_

### CODE REVIEW

File: parish/crates/parish-tauri/src/commands.rs [hunk 1]
```diff
@@ -1236,6 +1236,143 @@ pub async fn get_save_state(state: tauri::State<'_, Arc<AppState>>) -> Result<Sa
...
+pub(crate) async fn do_get_latest_screenshot(
+    state: &Arc<AppState>,
+) -> Result<Option<ScreenshotInfo>, String> {
+    let Some(path) = state.latest_screenshot_path.lock().await.clone() else {
+        return Ok(None);
+    };
+    let metadata = match std::fs::metadata(&path) {
+        Ok(m) => m,
+        Err(_) => return Ok(None),
+    };
```
[the remark this answers is not in the document]
Comment (dmooney) [reply parish/crates/parish-tauri/src/commands.rs]: Done — switched to `tokio::fs::metadata(&path).await` in c2d2955 so the stat doesn't block the Tokio executor when the MCP bridge is serving traffic. Thanks for the catch.

---
_Generated by [Claude Code](https://claude.ai/code)_
